File: frontend/src/types/customer.ts
export const CUSTOMER_SEGMENTS = [
  "enterprise",
  "smb",
  "trial",
  "consumer",
] as const;

export type CustomerSegment = (typeof CUSTOMER_SEGMENTS)[number];

export interface Customer {
  id: string;
  org_id: string;
  company_name: string;
  segment?: string;
  plan?: string;
  mrr?: number;
  arr?: number;
  account_manager?: string;
  renewal_date?: string;
  health_score?: number;
  industry?: string;
  employee_count?: number;
  created_at?: string;
  feedback_count?: number;
  avg_sentiment?: number;
}

export interface CustomerCreate {
  company_name: string;
  segment?: string;
  plan?: string;
  mrr?: number;
  arr?: number;
  account_manager?: string;
  renewal_date?: string;
  health_score?: number;
  industry?: string;
  employee_count?: number;
}

File: backend/app/schemas/customer.py
"""Pydantic schemas for customer API."""
from typing import Any

from pydantic import BaseModel, Field

CUSTOMER_SEGMENTS = ["enterprise", "smb", "trial", "consumer"]


class CustomerCreate(BaseModel):
    """Manual customer entry."""
    company_name: str = Field(..., min_length=1)
    segment: str | None = None
    plan: str | None = None
    mrr: float | None = None
    arr: float | None = None
    account_manager: str | None = None
    renewal_date: str | None = None
    health_score: int | None = Field(None, ge=0, le=100)
    industry: str | None = None
    employee_count: int | None = None


class CustomerOut(BaseModel):
    """Single customer for API response."""
    id: str
    org_id: str
    company_name: str
    segment: str | None
    plan: str | None
    mrr: float | None
    arr: float | None
    account_manager: str | None
    renewal_date: str | None
    health_score: int | None
    industry: str | None
    employee_count: int | None
    created_at: str | None
    feedback_count: int | None = None

    class Config:
        from_attributes = True


class CustomerListOut(BaseModel):
    """Paginated customer list response."""
    data: list[dict[str, Any]]
    total: int
    page: int
    per_page: int
    pages: int


class ImportCustomerRequest(BaseModel):
    """Import customer CSV request."""
    upload_id: str
    column_mapping: dict[str, str]


class ImportCustomerResponse(BaseModel):
    """Import customer result."""
    count: int
    segments: dict[str, int]
    total_arr: float
